src/Microsoft.AspNetCore.DataProtection.Extensions/DataProtectionProvider.cs

```diff
@@ -49,13 +53,14 @@ public static IDataProtectionProvider Create(DirectoryInfo keyDirectory)
             // build the service collection
             var serviceCollection = new ServiceCollection();
             var builder = serviceCollection.AddDataProtection();
-            builder.PersistKeysToFileSystem(keyDirectory);
 
-            if (setupAction != null)
+            if (keyDirectory != null)
```

Comment:
This works. It falls back to `DataProtectionServices.GetDefaultServices()` which contains all the default logic. Will add the null check.